@vates/parse-duration/src/index.ts

```diff
@@ -2,11 +2,11 @@
 
 const ms = require('ms')
 
-exports.parseDuration = value => {
+exports.parseDuration = (value: number | string): number => {
   if (typeof value === 'number') {
     return value
   }
-  const duration = ms(value)
+  const duration: number = ms(value)
```

Comment:
This looks wrong, `duration` can be a number as seen on the following line.

Reply:
I don't understand what you mean.

Reply:
My comment was incorrect, I fixed it.

See also: https://www.typescriptlang.org/play/?#code/OQVwzgpgBGAuBOBLAxrYAodATCyA2AhvNAGYgB2qiA9uVALZgAUi5ADiLAFwwKsDmUAD5RyIegCMI8AJQ8xk6cKgUcJVhCyYylWDTpsikACIh4BPbShMAbgTwgI88VPjK4ScvzmiXSgN7oUFCIJNawAJ5sENRhdg7QALzJUMAKrsAyUIHBwcSwZnTxjkFQAL6lyLRwUFhmFvrOim6JDMzFEDKlodZ15pZ0ya2qEOrkmlk5ubAAFvDUAO6iEEsAKlEQAKLw8-BMAAbk1LBQBFDxiFi19QM8ACT+HWX7XcEVeRAF8HR9DbToZSAA

Reply:
I found the d.ts about ms(), actually the function takes a string and return a number, or take a number and return a string!

```
/**
 * Short/Long format for `value`.
 *
 * @param {Number} value
 * @param {{long: boolean}} options
 * @return {String}
 */
declare function ms(value: number, options?: { long: boolean }): string;

/**
 * Parse the given `value` and return milliseconds.
 *
 * @param {String} value
 * @return {Number}
 */
declare function ms(value: string): number;

export = ms;

```

The compiler should has used the d.ts and showed the problems.

I have to rework on it.

Reply:
These community types are broken, `ms` can return `undefined` as we have seen in our own tests:

```js
ms('foo')
// → undefined
```